.github/workflows/pr_ci.yml

```diff
@@ -47,4 +47,13 @@ jobs:
         path: |
           test/OrchardCore.Tests.Functional/cms-tests/cypress/screenshots
           src/OrchardCore.Cms.Web/App_Data/logs
-  
+    - name: UI Tests
+      if: matrix.os == 'windows-latest'
+      run: |
+        dotnet test -c Release --no-restore --no-build ./test/OrchardCore.Tests.UI/OrchardCore.Tests.UI.csproj
+    - uses: actions/upload-artifact@v2
+      if: matrix.os == 'windows-latest' && failure()
```

Comment:
I don't think you need matrix.os == 'windows-latest' because if nothing is found on the path them the step still successfully completes with only a Warning: No files were found with the provided path: test/OrchardCore.Tests.UI/bin/Release/*/FailureDumps. No artifacts will be uploaded. warning message.

Reply:
I'd prefer not to have a warning when everything goes according to plan.

Reply:
You won't, if you only remove the `matrix.os == 'windows-latest'` like I said. If everything goes according to plan then the previous test is successful so `if: failure()` will evaluate to false.

Reply:
I see, good point.